feat(jsonc loader): add jsonc (#52)
nodejs-loaders/nodejs-loaders: packages/jsonc/README.md

# Nodejs Loaders: jsonc

<img src="https://raw.githubusercontent.com/nodejs-loader/nodejs-loaders/refs/heads/main/logo.svg" height="100" width="100" alt="@node.js loaders logo" />

> [![npm version](https://img.shields.io/npm/v/@nodejs-loaders/media.svg)](https://www.npmjs.com/package/nodejs-loaders/jsonc)

Reviewer: ```suggestion
[![npm version](https://img.shields.io/npm/v/@nodejs-loaders/media.svg)](https://www.npmjs.com/package/@nodejs-loaders/jsonc)
```

Reviewer: It looks like this is still incorrect?

Reviewer: Oh, weird. It is updated when I navigate from the PR

![unpacked size](https://img.shields.io/npm/unpacked-size/@nodejs-loaders/jsonc)

**Environments**: dev, test

**Compatible APIs**: [`module.register`](https://nodejs.org/api/module.html#moduleregisterspecifier-parenturl-options)

> To import JSONC files when using this loader, you will have to use `jsonc` as the file extension **and** in the [import attribute](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with).

Reviewer: ```suggestion
To import a JSONC file in node, it must have a `.jsonc` file extension **and** an [import attribute](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with) in the import statement (for consistency with Node.js's support for `json` imports):
```

```js

> import data from './data.jsonc' with { type: 'jsonc' };

Reviewer: ```suggestion
import data from './data.jsonc' with { type: 'jsonc' };

// OR

const data = await import('./data.jsonc', { with { type: 'jsonc' } });
```

```

```jsonc
{
  // JSONC file
  "key": "value"
  /* comment */
}
```

> 
> > [!IMPORTANT]
> > For consistency with Node.js's support for `json` imports, an import attribute is required for `jsonc` (case-sensitive: `jsonc`, not `JSONC`).

Reviewer: ```suggestion
```

Author: you want to remove it ?

Author: ping @JakobJingleheimer (need answers on that then we can publish it)

Reviewer: I think it's superfluous now because it's covered above. You can keep it if you want to extra emphasis it 🙂

Author: after re-read we can remove it because text already explain everything

# Nodejs Loaders: jsonc

<img src="https://raw.githubusercontent.com/nodejs-loader/nodejs-loaders/refs/heads/main/logo.svg" height="100" width="100" alt="@node.js loaders logo" />

> [![npm version](https://img.shields.io/npm/v/@nodejs-loaders/jsonc.svg)](https://www.npmjs.com/package/nodejs-loaders/jsonc)

Reviewer: ```suggestion
[![npm version](https://img.shields.io/npm/v/@nodejs-loaders/jsonc.svg)](https://www.npmjs.com/package/@nodejs-loaders/jsonc)
```

![unpacked size](https://img.shields.io/npm/unpacked-size/@nodejs-loaders/jsonc)

**Environments**: dev, test

> **Compatible APIs**: [`module.register`](https://nodejs.org/api/module.html#moduleregisterspecifier-parenturl-options), [`module.registerHooks`](https://nodejs.org/api/module.html#moduleregisterhooksoptions)

Reviewer: Unfortunately not

```suggestion
**Compatible APIs**: [`module.register`](https://nodejs.org/api/module.html#moduleregisterspecifier-parenturl-options)
```

To import JSONC files when using this loader, you will have to use `jsonc` as the file extension **and** in the [import attribute](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with).

```js
import data from './data.jsonc' with { type: 'jsonc' };
```

```jsonc
{
  // JSONC file
  "key": "value"
  /* comment */
}
```

> [!IMPORTANT]
> For consistency with Node.js's support for `json` imports, an import attribute is required for `jsonc` (case-sensitive: `jsonc`, not `JSONC`).

> > For consistency with Node.js's support for `json` imports, an import attribute is required for `jsonc` (case-sensitive: `jsonc`, not `JSONC`).

Reviewer: I think this got duplicated?

```suggestion
> For consistency with Node.js's support for `json` imports, an import attribute is required for `jsonc` (case-sensitive: `jsonc`, not `JSONC`).
```

> > [!WARNING]

Reviewer: ```suggestion
> [!IMPORTANT]
```

> The `jsonc` import attribute is required to use this loader and case-sensitive. So make sure to use `jsonc` and not `JSONC` or `Jsonc`.

> We do that because Node.js have same behavior for JSON file so we want to have same behavior for JSONC file.

Reviewer: ```suggestion
> For consistency with Node.js's support for `json` imports, an import attribute is required for `jsonc` (case-sensitive: `jsonc`, not `JSONC`).
```
